cpp/src/parquet/file_writer.cc

```diff
@@ -484,6 +516,22 @@ class FileSerializer : public ParquetFileWriter::Contents {
     }
   }
 
+  void WriteBloomFilter() {
+    if (bloom_filter_builder_ != nullptr) {
+      if (properties_->file_encryption_properties()) {
+        throw ParquetException("Encryption is not supported with bloom filter");
+      }
+      // Serialize page index after all row groups have been written and report
+      // location to the file metadata.
+      BloomFilterLocation bloom_filter_location;
+      bloom_filter_builder_->Finish();
+      bloom_filter_builder_->WriteTo(sink_.get(), &bloom_filter_location);
+      metadata_->SetBloomFilterLocation(bloom_filter_location);
+      // Release the memory for BloomFilter.
+      //      bloom_filter_builder_ = nullptr;
```

Comment:
Should we release this to avoid holding large memory?

Reply:
Why not?

Actually I think we'd better place bloom filters between row groups so we can proactively release the memory of serialized row groups as early as possible. But that can be a future optmization.

Reply:
> Actually I think we'd better place bloom filters between row groups so we can proactively release the memory of serialized row groups as early as possible. But that can be a future optmization.

We can support it later. I think it would be a bit tricky for RowGroup Split Handling. But it's ok for me. I can implement this firstly.